pkg/seal/seal.go

```diff
@@ -100,6 +101,16 @@ func (a *apiSealer) Raw(data Raw) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+func (a *apiSealer) Validate(ctx context.Context, secret io.Reader) error {
+	return kubeseal.ValidateSealedSecret(
+		ctx,
+		a.clientConfig,
+		a.ss.Namespace,
+		a.ss.Service,
```

Comment:
the e2e test is running with certURL https://github.com/bakito/sealed-secrets-web/blob/main/.github/workflows/e2e.yaml#L36
hence the ss.Namespace and ss.Service are empty

Reply:
other than OpenCert, the ValidateSealedSecret function of the sealed-secrets library does not accept a certURL parameter

Reply:
is this an error in the e2e test setup / config?

Reply:
The e2e test in main branch now runs the test against 1. in-cluster and 2. Cert-url setup.

#147

Reply:
If you merge the changes into your branch I'd expect the in-cluster tests to pass and cert-url tests zo fail

Reply:
are those defaults not applied in the e2e tests?
https://github.com/bakito/sealed-secrets-web/blob/50eaa618c3dc61b24bb8e3f5ca4215b7e57fa65b/chart/values.yaml#L21-L25

Reply:
Yes, these are the helm charts defaults unless overwritten

Reply:
in sealed-secrets api the certUrl will always be used if set,

```go
func OpenCert(ctx context.Context, clientConfig ClientConfig, controllerNs, controllerName string, certURL string) (io.ReadCloser, error) {
	if certURL != "" {
		return openCertLocal(certURL)
	}
```

so you don't need to implement that behavior in your config

Reply:
The issue with certURL is that it is only used if the sealed secrets service is not directly reachable as a cluster service.
As documented in the helm values https://github.com/bakito/sealed-secrets-web/blob/main/chart/values.yaml#L24

So this means, if someone uses the certURL, the target service is not reachable, which again will result in a failure of the validate function.

Reply:
What if the validation feature would be disabled if certURL is defined?

Reply:
On UI or both ?

Reply:
They might.
Until then, disabling it on the ui and returning nothing or a non 200 code on the api could do.

My guess is that 95% of installations go with in cluster config.

Reply:
There is a workaround for it which made it work for me cluster internal.
not sure if this is something which could be included in the helm chart